app/controllers/index.rb

```diff
@@ -0,0 +1,28 @@
+
+get '/' do
+  erb :index
+end
+
+get '/login' do
+end
+
+post '/login' do
+  user = User.find_by(username: params[:username])
+  if user.password == params[:password]
+    session[:user_id] = user.id
+    redirect "/#{user.id}"
+  else
+    redirect '/'
```

Comment:
when authentication fails it's nice to render the login form with an error message